src/sources/geojson.js

```diff
@@ -235,6 +235,7 @@ export class GeoJSONTileSource extends NetworkTileSource {
     prepareGeoJSON (data, tile, source) {
         // Apply optional data transform
         if (typeof this.transform === 'function') {
+            this.tile = tile;
```

Comment:
We should avoiding modifying `this` with transient data. But we also should avoid leaking `this` into these functions in the first place (an existing problem I hadn't realized!). It could open up some pretty crazy behavior, e.g. if modifications are then made in the function. I don't think we have this problem except in data source `transform` and `preprocess` functions; I can take care of these and will look for other possible cases.

For this case: to be more consistent with the existing syntax, I suggest making the `tile` an additional argument to the `transform` function:
`data = this.transform(data, this.extra_data, tile);`

If the calling code wants to access `tile` but doesn't have any `extra_data`, that's fine, it will just be an undefined value:

```
transform: |
    function(data, extra_data, tile) {
        t = tile;
        ...
```